scripts/asm-installer/release_asm_installer

```diff
@@ -158,20 +189,31 @@ write_and_upload() {
   local FILE_PATH; FILE_PATH="${BUCKET_PATH}/${FILE_NAME}"
   local FILE_URI; FILE_URI="${BUCKET_URL}/${FILE_PATH}"
 
-  if ! changes_necessary "${FILE_URI}" "${FILE_NAME}"; then return; fi
```

Comment:
I'm not sure why this got refactored, can you explain?

Reply:
I am not not sure if this is the best way to do it, but now the script has an option to write all the versions (with -a flag), as opposed to only write the versions that contains updates. Therefore, if there's no necessary changes, I have an `elif` clause to check if we would like to append the version regardless. Does this make sense?

Reply:
What use case are you imagining for that flag?

Reply:
I'm thinking the case when we want to create that file for the first time or re-create that file.  I did not assume that we already have that file so that's why I have this flag. If we could assume that we will never delete that file, like you commented above, then I think it's safe to not have this flag. Now that I think about it, the creation of the file is not complete anyway: it only has the version with the largest config number for each release. We probably will need some manual or separate scripting to backfill all the versions.

Reply:
Yep, I agree that a short script to initially backfill is a good idea. It's probably pretty small, but we should commit it just for transparency + code review.

Reply:
Will work on the script.